docs/ref/models/fields.txt

```diff
@@ -1641,6 +1641,22 @@ The possible values for :attr:`~ForeignKey.on_delete` are found in
     uses the primary key of the related object. If you reference a different
     field, that field must have ``unique=True``.
 
+    Note that ``to_field`` should points to local remote fields. For example
+    if you have below models::
+
+        class BaseModel(models.Model):
+            key = models.IntegerField(unique=True)
+
+        class ChildModel(BaseModel):
+            pass
+
+    You cannot have a relation like::
+
+        ref = models.ForeignKey(ChildModel, on_delete=models.CASCADE, to_field='key')
+
+    Because ``to_field`` points to the ``key`` field of ``BaseModel``. Unless
+    you change your ``BaseModel`` to an abstract model.
+
```

Comment:
Not sure it's worth a large mention in the docs as that's a pretty convoluted setup. I think a simple mention in the release notes about this backward compatibility change should be enough but I'll let @felixxm and @carltongibson decide.

Reply:
Agree, a small note in `Backwards incompatible changes in 3.1` should be enough.